packages/date-picker/docs/Calendar.stories.tsx

```diff
@@ -30,37 +31,97 @@ export default {
 const CalendarExample = (
   props: Partial<CalendarProps> & { id: CalendarProps["id"] }
 ): JSX.Element => (
-  <Calendar
-    mode="single"
-    weekStartsOn={DayOfWeek.Sun}
-    onDayChange={() => undefined}
-    defaultMonth={new Date(2022, 1, 5)}
-    locale={enAU}
-    {...props}
-  />
+  <div data-testid={props.id}>
```

Comment:
I was wondering why we couldn't just put this on `Calendar` itself (it's what I would expect) and it turns out we don't accept further props.

While we could certainly put it in now, the question is whether it should go on the container or the DayPicker. 
Ideally I'd expect to extend DayPicker because it's the main element, but it appears we can't actually remove the container because DayPicker doesn't accept `ref`, `id`, `role`, of which case it means `data-testid` should therefore go on the container.

Strange predicament here, but I suppose as a subcomponent it can be less flexible 🤷🏻 

TLDR; you can either leave it, or add `...restProps` to Calendar and put it in the container div (remember to use `OverrideClassName<HTMLAttributes<HTMLDivElement>>` if doing so).